### PR TITLE
Add note to pull.py

### LINKED ISSUE #69390: Running ansible-pull in parallel
##### SUMMARY
Executing two or more instances of ansible-pull in parallel without specifying the parameter "-d <DEST>, --directory <DEST>" results in undefined behaviour with race conditions.

##### ISSUE TYPE
- ~Bug  Report~
- Docs Report

##### COMPONENT NAME
ansible-pull

##### ANSIBLE VERSION
```
ansible 2.7.7
  config file = /home/user/.ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.7.3 (default, Apr  3 2019, 05:39:12) [GCC 8.3.0]
```

##### CONFIGURATION
```
[DEPRECATION WARNING]: DEFAULT_SUDO_FLAGS option, In favor of Ansible Become, which is a generic framework. See become_flags. , use become instead. This feature will be removed in version 2.8. Deprecation 
warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
ALLOW_WORLD_READABLE_TMPFILES(/home/user/.ansible.cfg) = True
ANSIBLE_NOCOWS(/home/user/.ansible.cfg) = True
ANSIBLE_SSH_ARGS(/home/user/.ansible.cfg) = -o ForwardAgent=yes
DEFAULT_FORKS(/home/user/.ansible.cfg) = 50
DEFAULT_GATHERING(/home/user/.ansible.cfg) = explicit
DEFAULT_HOST_LIST(/home/user/.ansible.cfg) = ['/home/user/inventories/common/hosts']
DEFAULT_ROLES_PATH(/home/user/.ansible.cfg) = ['/home/user/roles']
DEFAULT_SCP_IF_SSH(/home/user/.ansible.cfg) = True
DEFAULT_SUDO_FLAGS(/home/user/.ansible.cfg) = -i
DEFAULT_TRANSPORT(/home/user/.ansible.cfg) = smart
HOST_KEY_CHECKING(/home/user/.ansible.cfg) = True
RETRY_FILES_ENABLED(/home/user/.ansible.cfg) = False
```

##### OS / ENVIRONMENT
debian

##### STEPS TO REPRODUCE
```
any valid playbook
```

Execute:
```
ansible-pull --clean --track-subs --purge --full -U https://myrepository &
ansible-pull --clean --track-subs --purge --full -U https://myrepository &
ansible-pull --clean --track-subs --purge --full -U https://myrepository &
ansible-pull --clean --track-subs --purge --full -U https://myrepository &
```

Related lines of code:

https://github.com/ansible/ansible/blob/devel/lib/ansible/cli/pull.py#L124-L131

It would be good to document this as 

https://docs.ansible.com/ansible/latest/cli/ansible-pull.html

does not mention that in any way. It would be wise to introduce some kind of a check to see if the repository present in that folder is the one which is provided and abort in case of a missmatch.

It would be good to create an option to use a temporary folder instead of needing to create one by the calling process. This would eliminate the need to pass both arguments "--clean" and "--purge".

##### EXPECTED RESULTS
All playbooks should be executed.

##### ACTUAL RESULTS
Race conditions, different outputs. Here are some examples
```
 [ERROR]: Failed to remove /home/user/.ansible/pull/localhost: [Errno 2] No such file or directory: '/home/user/.ansible/pull/localhost'
```

```
localhost | FAILED! => {
    "changed": false,
    "msg": "Unable to determine hashes of submodules",
    "rc": 128,
    "stderr": "fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.\nUse '--' to separate paths from revisions, like this:\n'git <command> [<revision>...] -- [<file>...]'\nfatal: run_command returned non-zero status for roles/myrole\n.\n",
    "stderr_lines": [
        "fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.",
        "Use '--' to separate paths from revisions, like this:",
        "'git <command> [<revision>...] -- [<file>...]'",
        "fatal: run_command returned non-zero status for roles/myrole",
        "."
    ],
    "stdout": "Entering 'roles/myrole'\nHEAD\n",
    "stdout_lines": [
        "Entering 'roles/myrole'",
        "HEAD"
    ]
}
```

```
localhost | FAILED! => {
    "changed": false,
    "msg": "Failed to checkout branch master",
    "rc": 1,
    "stderr": "error: pathspec 'master' did not match any file(s) known to git\n",
    "stderr_lines": [
        "error: pathspec 'master' did not match any file(s) known to git"
    ],
    "stdout": "",
    "stdout_lines": []
}
```

### PR DESCRIPTION
**Pull request by saranti**
Fixes #69390 added note to PullCLI class.

##### SUMMARY
Added comment to PullCLI class documentation 

##### ISSUE TYPE

- Docs Pull Request

### CODE REVIEW

File: lib/ansible/cli/pull.py [hunk 1]
```diff
@@ -37,7 +37,8 @@ class PullCLI(CLI):
         each set to run via cron and update playbook source via a source repository.
         This inverts the default *push* architecture of ansible into a *pull* architecture,
         which has near-limitless scaling potential.
-
+        ``ansible-pull`` is not meant to run in parallel on the same host against the same checkout. You should either
+        add a lock (``setlock/flock``) around the invocation or use separate directories.
```
Comment (oraNod) [inline lib/ansible/cli/pull.py]:
```suggestion
        To make concurrent ``ansible-pull`` invocations for the same checkout on the same host, you should 
        use either locks (``setlock/flock``) or separate directories.
```

Comment (oraNod) [reply lib/ansible/cli/pull.py]: @saranti Thanks for the contribution! I've a suggestion to address the reader more directly. Hopefully it doesn't alter your original meaning.

**Comment by saranti** [comment]
No worries @oraNod. I added the changes you suggested

**Review by oraNod — APPROVED** [review]

File: lib/ansible/cli/pull.py [hunk 2]
```diff
@@ -37,7 +37,8 @@ class PullCLI(CLI):
         each set to run via cron and update playbook source via a source repository.
         This inverts the default *push* architecture of ansible into a *pull* architecture,
         which has near-limitless scaling potential.
-
+        To make concurrent ``ansible-pull`` invocations for the same checkout on the same host, you should
+        use either locks (``setlock/flock``) or separate directories.
```
Comment (bcoca) [inline lib/ansible/cli/pull.py:41]: this is very specific to your situation, I would generalize this:

'None of the CLI tools are designed to run concurrently with themselves, you should use an external scheduler and/or locking to ensure there are no clashing operations'

Even running from a different directory might create a concurrency problem with 2 instances trying to modify the same resources.

**Comment by samccann** [comment]
Closing in favor of https://github.com/ansible/ansible/pull/79387

**Pull request closed without merging**